build-tool-osx.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+#
+# This script installs the required build-time dependencies
+# and builds AppImageTool on OSX
+#
+
+small_FLAGS="-Os -ffunction-sections -fdata-sections"
+CC="cc -O2 -Wall -Wno-deprecated-declarations -Wno-unused-result"
+
+STRIP="strip"
+JOBS=${JOBS:-1}
+
+echo $KEY | md5sum
+
+set -e
+set -x
```

Comment:
See previous comment about this file, and the minimum changes I want to perrom to make it easy to synchronize with the original build.sh

Reply:
(There is no "previous comment".)

Again one of these "it was done this way elsewhere and I don't want to do it better because of it" arguments. Seriously, you don't have to worry about the other file. Just because we haven't got to fixing that as well, it doesn't mean you have to "keep this in sync"... Don't be frightened to do it properly here!

Reply:
Yes, I don't want to walk around and mess with other person's code, especially when I am a guest there and not the proper host, who takes decisions.

The main argument which was probably somewhere consumed and not stored properly was that, if I make extended changes there, then it would be very difficult to synchronize both scripts. Now the build.sh and build-tools-osx.sh file are "in sync" and it is pretty obvious which changes I had to do to make it compile with the BSD tools, instead of gnu. 

My fear is not of changing, is of maintaining. If you make any changes on the core script, it is very easy to find if these changes reflect to the other script as well. If I rewrite it from scratch would be more elegant, but who is going to update it on time?

Reply:
PS: do a diff between these two files and you'll see what I mean.

Reply:
I *know* what you mean...

Reply:
Also, when a maintainer gives suggestions, you shouldn't argue too much about staying compatible... especially when that maintainer tells you that this is absolutely not necessary...

Both files are going to be replaced with CMake, so no one really cares about staying compatible to the build script. Heck, we could even drop that file completely! But yes, these changes are just a necessity. If you feel better, you're free to synchronize both scripts.

If you now said "if it's going to replaced anyway, it won't matter" or something like that, I could accept it. But your whole point is that you are afraid of any maintenance problems. And considering that we maintainers even suggest you to make the changes, knowing about the incompatibility, you shouldn't feel this responsible for the maintenance. In the end it's us who will have to deal with this in the future. Don't worry, no one's going to blame you, they'd blame me, if any!

Reply:
That said, if you're capable of writing CMake scripts, we'd appreciate any contributions to the `CMakeLists.txt`. The build scripts are a bit too fragile anyway.

Reply:
I actually had also that in mind, that "since it was going to be replaced, why should I bother?", but I didn't feel OK to use this card. This card was also in the appimagetool.c code, with the #define you don't like, but again I don't feel that these arguments should be placed beforehand.

Unfortunately I don't know anything about CMake and the like - I even never understood neither had any luck with these tools.

So, since you gave me the freedom now, I would really say "these scripts are going to be replaced soon, and having something to look after would it be easier to produce the future CMake file probably for both platforms :) - and don't worth the effort any more".

Actually I asked in IRC yesterday: should I update the existing script, or create a new one based on the existing?

Reply:
@teras all right. I can't follow every discussion on IRC by the way, there's just too many of them at a time. Also, there's something called "real life" that sometimes consumes time. I'm okay with this point. I just wanted to point these suggestions out, as they are actual improvements. I invited you to discuss these with me, and I hope you still see that, albeit they won't be implemented, they're valid points.